### PR TITLE
Contributor guide

### PR DESCRIPTION
**Pull request by salsakran**
This is an old write up I had. Core committers should chat about this in the PR and then I'll add a link from the readme.

### CODE REVIEW

**Comment by salsakran** [comment]
And needless to say, Typos and ensuing Puns are firmly part of the Zen of Meta-bae's

**Comment by salsakran** [comment]
Added the contribution guide. Core committers, kindly chime in. 

I've made an effort to try to document our current process. This will be updated as things change, but there might be a bit of drift over time.

**Comment by salsakran** [comment]
Thinking we should have a set of labels under a common prefix that denote being in stages A-E.

Maybe 
* "stage/A-Disccusion"
* "stage/B-Scoping"
* "stage/C-Design"
* "stage/D-Implementation"
* "stage/E-Verfiication"
?

**Comment by agilliland** [comment]
For tracking the overall process I think we can use PRs as a simple way to handle D and E without having to worry as much about labels.  

Namely, once something gets the :+1: on the design and is approved to be worked on the next step should just be a PR (D) and I think it's fully understood that commentary about code and specific engineering choices will happen on the PR (E).  So as long as the PR properly references the original issue(s) we have a natural way to track the progress from design :arrow_right: implementation.

**Comment by salsakran** [comment]
Makes sense. 

Also, on a side note, I think it would be useful to start keeping track of Open Questions in the issue itself, rather than a separate doc. It would make it easier to get community feedback and hold onto a history of these decisions.

**Comment by salsakran** [comment]
Re: Open questions, this would probably be best done in a new issue, so that it could be the top, main description/comment.

So the flow would be:
* take one or more proposals/issues, create a new issue that references the old one
* The new issue would be the official one with a consistent format across all issues

thoughts?

File: docs/contributing.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+## Thank you
+
+First off, thanks for your interest in Metabase and for wanting to contribute!
+
+In this guide, we'll discuss how Metabase is built. This should give you a good sense of our process and where you might want to fit in.
+
+## What we're trying to build
+
+Metabase is all about letting non-technical users get access to the their organizations data. We're trying to maximize the amount of power that can be comfortably used by someone who understands their business, is quantitatively bent, but probably only comfortable with Excel.
```
Comment (mazameli) [inline docs/contributing.md]: > …access to ~~the~~ their **organization's** data

File: docs/contributing.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+To get a sense for the end goals, make sure to read the [Zen of Metabase](../zen)
+
+## Our product process:
+
+The core team runs a pretty well defined product process. It is actively being tweaked, but the below is a pretty faithful description of it at the time of writing. You should have a clear idea of how we work before jumping in with a PR. 
+
+### A) Identify product needs from the community
+
+We actively look for new feature ideas from our community, user base and our own use of Metabase internally. We concentrate on the underlying *problem* or *need*  as opposed to requests for specific features. While sometimes suggested features are built as request, often we find that they involve changes to existing features, and perhaps an entirely different solution to the underlying problem.
```
Comment (mazameli) [inline docs/contributing.md]: > …features are built as **requested**

File: docs/contributing.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+## Our product process:
+
+The core team runs a pretty well defined product process. It is actively being tweaked, but the below is a pretty faithful description of it at the time of writing. You should have a clear idea of how we work before jumping in with a PR. 
+
+### A) Identify product needs from the community
+
+We actively look for new feature ideas from our community, user base and our own use of Metabase internally. We concentrate on the underlying *problem* or *need*  as opposed to requests for specific features. While sometimes suggested features are built as request, often we find that they involve changes to existing features, and perhaps an entirely different solution to the underlying problem. 
+
+### B) Synthesis these needs into a concrete feature
```
Comment (mazameli) [inline docs/contributing.md]: ~~Synthesis~~ Synthesize

File: docs/contributing.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+This design is then handed off to one or more engineers.
+Together with the designer who owns the feature, they will work on the feature via a Pull Request.
+
+### E) Verification and merging
+
+If all goes well, the feature gets coded up, verified and then the pull request gets merged! High-fives all around.
+
+## Ways to help:
+
+The starting point would be to get familiar with Metabase the product, and know your way around. If you're using it at work, that's a great! If not, download the mac application or the jar and play around with it. Read the docs and generally get a feel for the flow of the product.
```
Comment (mazameli) [inline docs/contributing.md]: > that's ~~a~~ great!

That's a spicy meatball!

File: docs/contributing.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+If you want to help out, there are lots of ways. In order of increasing coordination + interaction with us:
+
+### Help with Identifying Needs and Problems Metabase can solve
```
Comment (mazameli) [inline docs/contributing.md]: Most other headings aren't Camel Case

File: docs/contributing.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+Some features, eg Database drivers, don't have any user facing pixels. These are a great place to start off contributing as they don't require as much communication, discussions about tradeoffs and process in general. 
+
+### Working on features that already have a visual design
+
+In situations where a design has already been done, we can always use some help. Chime in on a pull request or an issue and offer to help. 
+
+### #YOLO JUST SUBMIT A PR
+
+If tyou come up with something really cool, and want to share it with us, just submit a PR. If it hasn't gone through the above process, we probably won't merge it as is, but if it's compelling, we're more than willing to help you via code review, design review and generally OCD nitpicking so that it fits into the rest of our codebase.
```
Comment (mazameli) [inline docs/contributing.md]: > If ~~tyou~~ you come up…

**Comment by mazameli** [comment]
Holy context overload, github.

File: docs/contributing.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+This design is then handed off to one or more engineers.
+Together with the designer who owns the feature, they will work on the feature via a Pull Request.
+
+### E) Verification and merging
+
+If all goes well, the feature gets coded up, verified and then the pull request gets merged! High-fives all around.
+
+## Ways to help:
+
+The starting point would be to get familiar with Metabase the product, and know your way around. If you're using it at work, that's great! If not, download the mac application or the jar and play around with it. Read the docs and generally get a feel for the flow of the product.
```
Comment (tlrobinson) [inline docs/contributing.md]: Uppercase "Mac". Links to downloads?

**Pull request merged**